src/index.js

```diff
@@ -57,7 +60,9 @@ class MailchimpSubscribe extends React.Component {
 
 MailchimpSubscribe.propTypes = {
   render: PropTypes.func,
-  url: PropTypes.string.isRequired
+  url: PropTypes.string.isRequired,
+  onSuccess: PropTypes.func,
+  onError: PropTypes.func,
```

Comment:
Those are missing `defaultProps` below. For functions, it's a good practice to give a default value of a noop `() => {}` function.

Then, you won't need the `if(this.props.onSuccess!==undefined)` checks anymore.